sysutils/scp-backup/src/opnsense/scripts/OPNsense/ScpBackup/ScpBackup.php

```diff
@@ -0,0 +1,65 @@
+#!/usr/local/bin/php
+<?php
+
+/**
+ *    Copyright (C) 2018 David Harrigan
+ *    Copyright (C) 2015 - 2017 Deciso B.V.
+ *
+ *    All rights reserved.
+ *
+ *    Redistribution and use in source and binary forms, with or without
+ *    modification, are permitted provided that the following conditions are met:
+ *
+ *    1. Redistributions of source code must retain the above copyright notice,
+ *       this list of conditions and the following disclaimer.
+ *
+ *    2. Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *
+ *    THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ *    INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ *    AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *    AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ *    OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+require_once("util.inc");
+require_once("config.inc");
+
+use OPNsense\Base;
+use OPNsense\Core\Config;
+
+$config = Config::getInstance()->object(); 
+$scpBackup = $config->OPNsense->ScpBackup;
+
+if(isset($scpBackup) && isset($scpBackup->enabled) && $scpBackup->enabled == 1) {
+    $hostname = $scpBackup->hostname;
+    $port = empty(trim($scpBackup->port)) ? "22" : $scpBackup->port;
+    $username = empty(trim($scpBackup->username)) ? "root" : $scpBackup->username;
+    $remoteLocation = empty(trim($scpBackup->remoteLocation)) ? "./" : $scpBackup->remoteLocation;
+    $identifyFile = "/conf/sshd/ssh_host_rsa_key";
+
+    if(!substr($remoteLocation, -1) == "/") {
+        $remoteLocation = $remoteLocation . "/";
+    }
+
+    $remoteLocationFullPath = $remoteLocation . "config-" . date('Y-m-d-H-i') . ".xml";
+    $command = "scp -P $port -i $identifyFile /conf/config.xml $username@$hostname:$remoteLocationFullPath";
```

Comment:
Looks like RCE

Reply:
I don't believe it is - can you show if it is? Happy to change if it's provable. I do escape the command as per PHP guidelines.

Reply:
You are using data from config.xml unescaped in the shell command - escape them with https://secure.php.net/manual/en/function.escapeshellarg.php

the variables that need escaping are 
* $username
* $hostname
* $remoteLocationFullPath

Reply:
Fantastic! Thank you. I'll do that :-)

-=david=-